repo-settings.c

```diff
@@ -15,43 +15,43 @@ void prepare_repo_settings(struct repository *r)
 	/* Defaults */
 	memset(&r->settings, -1, sizeof(r->settings));
 
-	if (!repo_config_get_bool(r, "core.commitgraph", &value))
+	if (!repo_config_get_bool(r, "core.commitGraph", &value))
 		r->settings.core_commit_graph = value;
-	if (!repo_config_get_bool(r, "gc.writecommitgraph", &value))
+	if (!repo_config_get_bool(r, "gc.writeCommitGraph", &value))
```

Comment:
Don't do this. The established pattern in the config code is to use lower case. I'm pretty sure that's necessary for correctness.

Reply:
Yep. It won't work unless it is all lower-case.

Reply:
BTW this probably means that test coverage for this feature is missing because the test suite does not catch this.

Reply:
Okay, I will drop this commit. 

I thought it would be a bit easier to grep (since the documentation all use camel case) but turns out this is a hard requirement. 🤣

Reply:
Well, as you found this issue, probably you want to add a regression test to verify that `gc.writeCommitGraph` actually works.

Reply:
> BTW this probably means that test coverage for this feature is missing because the test suite does not catch this.

I _guarantee_ you that there are tests for at least half of the changed config keys yere.

Apparently, the config code _does_ [normalize the strings to lower-case](https://github.com/git/git/blob/efcab5b7a3d2ce2ae4bf808b335938098b18d960/config.c#L411) before checking the hashmap, so this isn't broken. So, correctness is not an issue. However, I believe using lowercase literals is the convention in the rest of the codebase. [The coding guidelines](https://github.com/git/git/blob/efcab5b7a3d2ce2ae4bf808b335938098b18d960/Documentation/CodingGuidelines#L513-L531) mention camel case for the _documentation_ of config variables, but does not mention the casing locally.

Reply:
Personally, as a new contributor, I do find it easier for the code to be consistent with the doc. That way instead of `grep -ir writeCommitGraph` I can just `grep -r writeCommitGraph`. 🤔 

It seems like this is more of a convention issue than correctness. So how about we leave the commit there, let the PR be sent to the git devel mailing list and voice opinion to the group there?

Reply:
You can try that, but I will personally say "this commit is not helpful." If you were to instead make a concerned effort to make this casing consistent throughout the codebase AND update the `CodingGuidelines` document, then that would be a different series with a different purpose.

Inserting this case change here distracts from the goal of your series.

Reply:
> So how about we leave the commit there, let the PR be sent to the git devel mailing list and voice opinion to the group there?

Whenever I read a comment like this, I feel that the reviewer's expertise was dismissed. I am sure you did not want to sound like that, but it sure sounds like that to me.

Reply:
@derrickstolee I think your suggest is quite clear here. I will consider it for another proposal for `CodingGuidelines`. Although I wonder if it is worth the churn/noise at all, given that my recent patch to remove the 'dollar-sign inside artihmetic expansion' were turned down for same reason.

@dscho Haha you are right! By no mean I claim to be an expert here. All of these are good learning experience for me, please keep the feedbacks coming. 👍